### ISSUE TITLE
Hysteria Optimization

### ISSUE THREAD

**Issue by private-number** [apernet/hysteria#644]
### Discussed in https://github.com/apernet/hysteria/discussions/641

<div type='discussions-op-text'>

<sup>Originally posted by **private-number** August  4, 2023</sup>
Hi guys

Consider me as newbie and ELI5 

I was looking to use hysteria but I need help how to find best "recv_window_conn" and "recv_window" and "up_mbps" and  "down_mbps"

I think up/down numbers are about what is my download speed when i download/upload directly so i need to know if i'm wrong or right about it.

about "recv_window_conn" and "recv_window" i read that it's different for everyone and I need to find optimum number in order to reach hysteria full capability and speed. but I don't have any idea how to find it.

Thanks.</div>

**Comment by chika0801** [comment]
https://hysteria.network/docs/optimizations/

**Comment by private-number** [comment]
> https://hysteria.network/docs/optimizations/
Thanks for your reply.
I already read that but i couldn't figure it out.
It would be so much help to explain how to find perfect numbers

**Comment by chika0801** [comment]
You ping your VPS host's IP from your own computer, and if you get a ping value of 150ms. the download speed between you and your VPS host, after your own testing, assuming that it can stabilise at 80MB/s, 80MB = 80X8 480Mbps.

At this point, according to Hy's instructions, the recommended algorithms for sysctl -w net.core.rmem_max= and sysctl -w net.core.wmem_max= are 150ms = 0.15s , 0.15s X 80 MB/s = 12MB.

12 X 1024 X 1024 = 12582912

So

sysctl -w net.core.rmem_max= 12582912
sysctl -w net.core.wmem_max= 12582912
